### PR TITLE
Feat/avoid repetitive values when the filter is day and value does not change 259

### LINKED ISSUE #259: Avoid repetitive values when the filter is `day` and value does not change
Image

### PR DESCRIPTION
**Pull request by JAMares**
### What does this PR do?

- Resolve #259

### Steps to test

1. Run `yarn storybook`
2. Check performance

Now:

https://github.com/defactor-com/ui-kit/assets/49013556/7a2454a9-c845-4aa8-aa36-3450d1ea4154

#### CheckList

- [X] Follow proper Markdown format
- [X] Content is available in all languages
- [X] The content is adequate
- [X] I Ran a spell check
- [X] I included test coverage

### CODE REVIEW

**Review by AngeloCG97 — CHANGES_REQUESTED** [review]
This space is necessary?

File: src/components/LineChart/useLineChartState.tsx [hunk 1]
```diff
@@ -61,8 +93,28 @@ const useLineChartState = ({ data, series }: ILineChartState) => {
 
   const missingData = !data?.length || !series.length;
 
+  useEffect(() => {
+    const hideSeries = Object.keys(hide).filter((clave) => hide[clave]);
```
Comment (AngeloCG97) [inline src/components/LineChart/useLineChartState.tsx:97]: Please add space before if condition

File: src/components/LineChart/useLineChartState.tsx [hunk 2]
```diff
@@ -61,8 +93,28 @@ const useLineChartState = ({ data, series }: ILineChartState) => {
...
+    if (series.length - hideSeries.length === 1) {
+      const value = series.filter((serie) => !isHide(serie.name))[0];
+      const isSameData = validatedSeries.filter(
+        (vSerie) => vSerie.name === value.name
+      )[0].isSameData;
+      if (isSameData) setTickCount(1);
```
Comment (AngeloCG97) [inline src/components/LineChart/useLineChartState.tsx:103]: same

File: src/stories/LineChart.stories.tsx [hunk 3]
```diff
@@ -125,7 +133,7 @@ const Template: Story<ILineChart> = (args) => {
 
 export const LineChartItem = Template.bind({});
 LineChartItem.args = {
-  formatValue,
+  // formatValue,
```
Comment (AngeloCG97) [inline src/stories/LineChart.stories.tsx:136]: this comment is necessary?

File: src/stories/LineChart.stories.tsx [hunk 4]
```diff
@@ -14,47 +15,54 @@ export default {
...
+const formatValue: FormatValueType = (
   value: number | string,
   options?: Intl.NumberFormatOptions
 ) => {
-  return value.toLocaleString("en-US", {
-    style: "currency",
-    currency: "USD",
-    ...options,
-  });
+  let numeroStr = value.toString();
```
Comment (xavier506) [inline src/stories/LineChart.stories.tsx:22]: @JAMares  can we stick to english names for variables please

Comment (JAMares) [reply src/stories/LineChart.stories.tsx]: Fixed it

File: src/stories/LineChart.stories.tsx [hunk 5]
```diff
@@ -14,47 +15,54 @@ export default {
...
   options?: Intl.NumberFormatOptions
 ) => {
-  return value.toLocaleString("en-US", {
-    style: "currency",
-    currency: "USD",
-    ...options,
-  });
+  let numberStr = value.toString();
+
+  let firtsDigits = numberStr.slice(0, 2);
```
Comment (xavier506) [inline src/stories/LineChart.stories.tsx:24]: first not firts

**Review by AngeloCG97 — APPROVED** [review]

**Pull request merged**